### DISCUSSION TITLE
24 September 2024 2200 UTC

### DISCUSSION THREAD

**Discussion by ktoddbrown** [ktoddbrown/SoilDRaH#52] [RegularCalls]
### Agenda

These calls are lead by @ktoddbrown and are generally open calls.
Please contact them for call in information.
The purpose of these meetings are to move forward current issues related to the development of Soil-DRaH.
This may include teaching, mutual accountability, code review, paired programming, or other activities.

## (5 | 2200) Check ins
- Who is here
- Note taking?
- How are you showing up? <https://feelingswheel.com/>

## (20 | 2205) *Formulation* Variable mapping
- Review how variables might be mapped by walking through  [Curate ISCN4](https://github.com/ktoddbrown/SoilDRaH/blob/28-iscn4/vignettes/articles/Curate_ISCN4.qmd)

## (20 | 2225) Issues checkins and stuck to unstuck
- *Pull up open issues where do each issue stand?*
- *How can we help each other move from stuck to unstuck on a topic?*

## (5 | 2250) Backlog
- *What new issues need to be created?*

**Comment by vaasukimarupaka** [comment]
### (5 | 2200) Check ins
Who is here Gabriel, Maggie, Graham, KTB, Vaasuki, Brandon, Courtney
Note taking? Maggie 
How are you showing up? https://feelingswheel.com/
### (20 | 2205) Formulation Variable mapping
Review how variables might be mapped by walking through [Curate ISCN4](https://github.com/ktoddbrown/SoilDRaH/blob/28-iscn4/vignettes/articles/Curate_ISCN4.qmd)
start talking about integration - align variables across datasets
branch 28 for follow along
vignettes>articles>Curate_ISCN4
libraries - tidyverse, RSQLite, knitr, kableextra, maps
echo=FALSE - doesn't print
ISCN map - study, source, target variable, target method, action note (things that need handcoding)
Units? yes, in metadata - some transformations in action notes
source_variable=of_variable in readscripts for now
controlled vocab coming for target variable in metadata (ISCN4 annotations)
primary_variables - variables to keep in main environment (memory management)
read in as level 0 (long format)
datasets themselves not pushed to repo
read in, give variables, remove NRCS(changed set), inner join with annotations, join_by - changes to of_variable
multiple methods separated by :
what do levels imply? amount of modification we've done - level0 is long read, level 1 is subset of original file (this used to link in)
Methods? multiple can be used when some are more generalized and others are specific
### (20 | 2225) Issues checkins and stuck to unstuck
Pull up open issues where do each issue stand? - What are we working on?
Vaasuki - NCSS read functions - looks fine, works, when running (if verbose true) sometimes runs into infinite loop (in read function on line 22)
try to identify bug and fix it, why verbose causing issues? 
line by line would be a possible fix (why hanging on a specific line?)
Maggie - ISCN3 and FIA running the code, maybe work on Find data in wiki
Courtney - finding out how to work together
Brandon - still looking into what things mean
Graham - report longleaf pine, fixed filenames and units
Gabriel - updated issue 42 (wiki to-dos from monday meeting)
subissues - how tos for readscripts and annotations
open questions? no

**How can we help each other move from stuck to unstuck on a topic?**

Graham questions: 
columns as flag identifiers - cross referencing/merging for data - go with overlapping variables; same identifier, treat the same
of_variable: sample -> stand - some not showing up in post columns (are we missing columns from annotations)
Maggie Questions - verbose meaning - print me messages - 
used here as a flag for understanding functions
is this specific to this lab? variable in R functions - is a prog practice
tells about what the function is doing more in depth (warnings mean things will work but you don't want it, error stops the code)
### (5 | 2250) Backlog
What new issues need to be created?
### (5 | 2255) Check outs [appreciations, discoveries, and ToDo]
